### DISCUSSION TITLE
Timeout when using with AWS EKS

### DISCUSSION THREAD

**Discussion by Dunge** [helmfile/helmfile#776] [General]
I used helmfile locally on a microk8s cluster without errors for a while, now I'm using the exact same solution over AWS and the helm Diff plugin seems to break like half of the time with timeout errors.

```
  Error: Failed to get release kube-prometheus-stack in namespace monitoring: exit status 1: E0331 14:19:52.253059   38064 request.go:1058] Unexpected error when reading response body: http2: client connection lost
  Error: query: failed to query with labels: unexpected error when reading response body. Please retry. Original error: http2: client connection lost
  Error: plugin "diff" exited with error
```

or

```
  Error: Failed to get release kube-prometheus-stack in namespace monitoring: exit status 1: E0331 14:18:21.540261   29552 request.go:1058] Unexpected error when reading response body: read tcp 10.1.80.48:59528->54.161.107.218:443: wsarecv: An existing connection was forcibly closed by the remote host.
  Error: query: failed to query with labels: unexpected error when reading response body. Please retry. Original error: read tcp 10.1.80.48:59528->54.161.107.218:443: wsarecv: An existing connection was forcibly closed by the remote host.
  Error: plugin "diff" exited with error
```

Running the operation multiple time at some point it get through, but it's extremely annoying.

Anyone have any idea what could be causing this?

**Comment by yxxhero** [comment]
@Dunge please your command. and maybe it's your network issue.

**Reply by Dunge** [reply]
The command is simply a `helmfile apply`.  The network is a very fast 1.5gbps corporate connection and I never had issues of the genre with it.

I saw #682 is reporting the exact same issue.

I managed to work around by splitting my project into two different helmfile. One with monitoring charts like prometheus/loki/grafana that I don't update often and seemed to be heavy when doing the diff operations and another one for my actual application that have less ressources but I update frequently. This way I seem to avoid the problem, but I'm pretty sure there is one and it's not my network.

**Comment by yxxhero** [comment]
@Dunge thanks for your feedback. And I wonder to know the number of your k8s resource in your release.
 BTW, you can post an issue in the helm-diff repo too.  https://github.com/databus23/helm-diff
